fish_prompt.fish

```diff
@@ -861,13 +861,31 @@ function __bobthefish_prompt_desk -S -d 'Display current desk environment'
 end
 
 function __bobthefish_prompt_nvm -S -d 'Display current node version through NVM'
-    [ "$theme_display_nvm" = 'yes' -a -n "$NVM_DIR" ]
```

Comment:
I'm not sure about `nodenv`, but running `nvm` every time we render a prompt is _slow_. That's why it's opt-in rather than opt-out, and that's also why we check `$NVM_DIR` and bail early, rather than calling `nvm current`.

I'd love to support `nodenv`, but I really don't want to slow down the prompt for everyone all the time to do it :)

Reply:
Thanks for responding, @bobthecow! :smiley: 

> running nvm every time we render a prompt is slow. That's why it's opt-in rather than opt-out, and that's also why we check $NVM_DIR and bail early, rather than calling nvm current.

That makes sense to me. Won't the new check/guard (right below this) ensure that we fail/bail early? :thinking: 

```
 [ "$theme_display_nvm" = 'no' ]
    and return
```

Also, I can add the check for `$NVM_DIR` to the `if type -fq nvm` line if that works?

Let me know what you think! :smile:

Reply:
My point was that expensive checks should always be opt-in, not opt-out. And I'm not sure about `nodenv`, but `nvm` is _definitely_ expensive, so it shouldn't be on by default.

Yes, the check for `$NVM_DIR` should be preserved. Maybe add it to the `nvm` branch, something like…

```
if type -fq nvm
  [ -n "$NVM_DIR" ]
  or return

  # ...
```

Reply:
> Yes, the check for `$NVM_DIR` should be preserved. Maybe add it to the `nvm` branch, something like…
> 
> ```
> if type -fq nvm
>   [ -n "$NVM_DIR" ]
>   or return
> 
>   # ...
> ```

I updated this to what you suggested. :smiley: